File: 정렬/선택정렬/선택정렬.js
const array = [7, 5, 9, 0, 3, 1, 6, 2, 4, 8];

for (let i = 0; i < array.length; i++) {
  let min_index = i;
  for (let j = i + 1; j < array.length; j++) {
    if (array[min_index] > array[j]) {
      min_index = j; //전체 비교를 한 번 할때마다 최솟값인덱스를 저장해줌
    }
  }
  let temp = array[min_index];
  array[min_index] = array[i];
  array[i] = temp;
}

console.log(array);

File: 정렬/선택정렬/선택정렬.py
array = [7,5,9,0,3,1,6,2,4,8]

for i in range(len(array)): # 0~9, 전체반복
    min_index = i #가장 작은 원소의 인덱스
    for j in range(i+1, len(array)): 
        if array[min_index] > array[j]: #비교값이 더 작은값으면
            min_index = j #min_index를 비교값 인덱스로 설정(더 작은값이 됨)
    #타겟값과 전체 원소의 비교가 끝나면 min_index = array[최솟값]
    array[i], array[min_index] = array[min_index], array[i] #타겟값과 자리바꿈
print(array)